Cabal/Distribution/Simple/Build/PathsModule.hs

```diff
@@ -67,6 +68,16 @@ generate pkg_descr lbi clbi =
           "import System.Environment (getExecutablePath)\n"
         | otherwise = ""
 
+       overloaded_strings_imports
+         | supports_overloaded_strings =
+           "import Data.String (fromString)\n"
```

Comment:
Tbh, I'd prefer if we went for different way here. I can imagine a couple of ways to break this fragile workaround. But there's a simpler way to accomplish this w/ less assumptions about the compilation scope/environment: Just use `{-# LANGUAGE NoOverloadedStrings #-}` to neutralise `default-languages: OverloadedStrings` locally.

Reply:
I'd also like to point out that overloaded lists has the same effect, so you'd have to add both suppressors.

Reply:
Alternately stopping rebindable syntax seems like the best bet.

Reply:
Alternately stopping rebindable syntax seems like the best bet.

Reply:
@hvr Ah, thanks for pointing that out!  I'm sure you could break this if you wanted to. I had indeed forgotten about `-XNo*`, since I never use it other than for `NoImplicitPrelude`.  For example:

```haskell
{-# LANGUAGE NoWeirdStuff #-}
```

causes

```
hmm.hs:1:14: error: Unsupported extension: NoWeirdStuff
```

So, this will need to be conditional on compilation scope/environment regardless.  I think there should be a pragma to reset language extensions.  I have opened https://ghc.haskell.org/trac/ghc/ticket/14685

Reply:
There is also precedent for my original approach.  Why do you think the generated code has `import Prelude`??  I will also switch that over to `{-# ImplicitPrelude #-}`.

@rohit507 `OverloadedLists` is not a problem on its own, because it only needs `fromListN` to be in scope if `RebindableSyntax` is enabled.

Reply:
Indeed, I'm not happy with the current approach either; The generated `Paths_` module should try to make as little assumptions about the module namespace as possible and this has been a pet peeve of mine anyway; but there's no reason to increase the technical debt now; and I think there's better ways to accomplish the goal without making things worse.

Reply:
fwiw, there's a general pattern here: Starting with GHC 7.0, Cabal can emulate a resetting `-X` flag: it just needs to neutralise (almost) all  `-X` flags (and Cabal knows enough about them to know which have a `No` counterpart). Then we could easily also avoid the redundant `import Prelude`.

Reply:
We only need this for `Paths_*.hs`